sdk/core/core/test/cmocka/test_az_span.c

```diff
@@ -751,6 +751,14 @@ static void az_span_u32toa_overflow_fails(void** state)
   assert_true(az_span_u32toa(buffer, v, &out_span) == AZ_ERROR_INSUFFICIENT_SPAN_SIZE);
 }
 
+static void az_span_copy_empty(void** state)
```

Comment:
While you are adding tests here, can you please add cases where both source and destination are empty as well.

Reply:
not relevant to this PR. 
That should be part of a different issue that would cover lack of coverage for az_span_copy, of something like that.

here we only care about managing empty source without considering destination